mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/control/plane/resource/cluster/proto/MantisResourceClusterSpec.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.control.plane.resource.cluster.proto;
+
+import io.mantisrx.shaded.com.fasterxml.jackson.annotation.JsonCreator;
+import io.mantisrx.shaded.com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Map;
+import java.util.Set;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+import lombok.Singular;
+import lombok.Value;
+
+/**
+ * Contract class to define a Mantis resource cluster. This contract provides the abstraction to provide a generic
+ * definition from Mantis control perspective, and it's up to the implementations of each
+ * {@link io.mantisrx.control.plane.resource.cluster.resourceprovider.IResourceClusterProvider} to translate this spec
+ * to corresponding framework's cluster/node(s) definition.
+ */
+@Value
+@Builder
+public class MantisResourceClusterSpec {
+
+    @NonNull
```

Comment:
Let's not use the @NonNull annotation. By default, let's assume all fields are non-null by default. If something is nullable, use optional types in case that class is in the API layer (i.e., external-facing) or @Nullable if the field is only within a class (i.e., internal-facing).